tests/common/helpers/parallel.py

```diff
@@ -104,6 +103,34 @@ def force_terminate(workers):
                         """Processes running target "{}" could not be terminated.
                         Unable to kill {}:{}, error:{}""".format(target.__name__, p.pid, p.name, err)
                     )
+                finally:
+                    # If sanity check is timeout and killed, we still want to put some information in results.
+                    # process name looks like _check_bgp_on_dut--<MultiAsicSonicHost sonic>
+                    # parse it to get sanity check item and hostname
+                    # For other processes, just return a dict {'failed': True}.
+                    # node looks like <EosHost VM0100> or <MultiAsicSonicHost sonic>
+                    node = p.name.split('--')[1]
+                    node_items = node.split()
+                    if len(node_items) > 1:
+                        node_name = node_items[1][:-1]
+                    else:
+                        node_name = node_items[0]
```

Comment:
This way of getting the node_name is too tricky and not robust enough. We can't guarantee that the node looks like <EosHost VM0100> or <MultiAsicSonicHost sonic>.
Since the "p" is an instance of SonicProcess, is it possible to add an attribute to SonicProcess for storing node name?

Reply:
@wangxin Per discussion, will submit a new PR to cover this part.